components/storage/blockdevice/COMPONENT_SPIF/SPIFBlockDevice.cpp

```diff
@@ -146,56 +138,39 @@ int SPIFBlockDevice::init()
         tr_debug("Initialize flash memory OK");
     }
 
-    /* Read Manufacturer ID (1byte), and Device ID (2bytes)*/
-    spi_status = _spi_send_general_command(SPIF_RDID, SPI_NO_ADDRESS_COMMAND, NULL, 0, (char *)vendor_device_ids,
-                                           data_length);
-    if (spi_status != SPIF_BD_ERROR_OK) {
-        tr_error("init - Read Vendor ID Failed");
+    if (_handle_vendor_quirks() < 0) {
+        tr_error("Init - Could not read vendor id");
         status = SPIF_BD_ERROR_DEVICE_ERROR;
         goto exit_point;
     }
 
-    switch (vendor_device_ids[0]) {
-        case 0xbf:
-            // SST devices come preset with block protection
-            // enabled for some regions, issue global protection unlock to clear
-            _set_write_enable();
-            _spi_send_general_command(SPIF_ULBPR, SPI_NO_ADDRESS_COMMAND, NULL, 0, NULL, 0);
-            break;
-    }
-
     //Synchronize Device
     if (false == _is_mem_ready()) {
         tr_error("init - _is_mem_ready Failed");
         status = SPIF_BD_ERROR_READY_FAILED;
         goto exit_point;
     }
 
-    /**************************** Parse SFDP Header ***********************************/
-    if (sfdp_parse_headers(callback(this, &SPIFBlockDevice::_spi_send_read_sfdp_command), _sfdp_info) < 0) {
-        tr_error("init - Parse SFDP Headers Failed");
-        status = SPIF_BD_ERROR_PARSING_FAILED;
-        goto exit_point;
-    }
-
+    /**************************** Parse SFDP headers and tables ***********************************/
+    {
```

Comment:
Why is this set of scope brackets here ?

Reply:
Only reason I added those was to show clearly which part of the code has something to do with SFDP parsing. And to remove comments which were saying the same thing as functions being called.